burble_sort_by.rb

```diff
@@ -0,0 +1,20 @@
+def bubble_sort_by(arr)
+  num_of_sort = 0
+  length_of_arr = arr.length - 1
+  length_of_arr.times do
+    (0..arr.length - 2 - num_of_sort).each do |i|
+      left = arr[i]
+      right = arr[i + 1]
+      next unless yield(left, right).positive?
+
+      temp = arr[i]
+      arr[i] = arr[i + 1]
+      arr[i + 1] = temp
+    end
+    num_of_sort += 1
+  end
+  puts arr
+end
+bubble_sort_by(%w[allof me those you them]) do |left, right|
+  left.length - right.length
+end
```

Comment:
- From line 18 to line 20 are debugging code. Please delete it or keep it on another branch. 
- Erase the `puts` on line 16 so it will be only `arr`. This `puts` is causing the code to fail.